### ISSUE TITLE
questions about UART driver usage

### ISSUE THREAD

**Issue by adamswsk** [edgexfoundry/device-uart#45]
Hello, I have deployed edgex-uart using edgex-compose with the following command:

$ make gen ds-uart no-secty ui

Subsequently, in the Edgex UI, I modified the Uart-Generic-Device settings to baudRate=9600 and deviceLocation=/dev/ttyUSB0. When I attempted to send a command using the Command: String, set method, it returned a code of 503, with the message: failed to send a http request.

My questions are:

Could this issue be caused by the Docker container not having root permissions on Ubuntu? This is because I can control the device using the following methods:

$ set serial -g /dev/ttyUSB0
$ stty 9600 -F /dev/ttyUSB0
$ echo -e -n ‘\x99\x03\x0c\x74\x24\x51\x99’>/dev/ttyUSB0

Additionally, I need to send a hexadecimal array; can the String method be used for this purpose? I've tried the int8 method but it only allows sending one byte of data at a time. If the String method is not suitable, would it suffice to modify the valueType in the Device Profile? What value should it be changed to?

**Comment by cloudxxx8** [comment]
the project contributor may not read Chinese, so please write down your question in English
also, you may check the logs from ui, core-command, and device-uart to get more information

**Comment by adamswsk** [comment]
I modified the generated docker-compose.yml file to add a volume mount in the device-uart section: - /dev/ttyUSB0:/dev/ttyUSB0, and then accessed the container's shell using sudo docker exec -u root -it container_id /bin/sh. Inside the shell, running ls -l /dev/ttyUSB0 returned crw-rw---- 1 root dialout 188, 0 Mar 7 06:19 /dev/ttyUSB0. However, when I attempted to set the baud rate with stty 9600 -F /dev/ttyUSB0, it resulted in the error message 'Operation not permitted'.
